### PR TITLE
don't crash when you lose all your spawns

### PR DESCRIPTION
**Pull request by karlthepagan**
previous solution causes an undef, see https://github.com/ScreepsOCS/screeps.behaviour-action-pattern/pull/956/files

### CODE REVIEW

**Comment by democratic-collaboration** [comment]
[democratic collaboration](https://github.com/TooAngel/democratic-collaboration)
Approved reviews will speed up the merge, request changes will slow it down.

Please review the PR to help.

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 165/336
    Coefficient: 0.491071428571
    Merging in 5 days 0 hours
    Age 0 days 1 hours

File: src/config_brain_memory.js [hunk 1]
```diff
@@ -188,7 +188,7 @@ Upgrade less: ${strings.upgradeLess}
 };
 
 brain.prepareMemory = function() {
-  Memory.username = Memory.username || _.find(Game.spawns, 'owner').owner.username;
+  Memory.username = Memory.username || _.chain(Game.rooms).map('controller').flatten().filter('my').map('owner.username').first().value();
```
Comment (Nevrdid) [inline src/config_brain_memory.js]: hum, shoundn't crash as it is save in memory, and didn't get error destroying it. but if you want, this should be lighter :
`_(Game.rooms).find(room => room.controller.my).value().controller.owner.username`

Comment (karlthepagan) [inline src/config_brain_memory.js]: Depending on the state of memory is not failsafe. This change fixes that.

Comment (karlthepagan) [inline src/config_brain_memory.js]: Vision to a highway or center nine room will crash on memory clear with your alternative.

Comment (Nevrdid) [inline src/config_brain_memory.js]: didn't understood what you mean about memory.
`_.find(Game.rooms, room => room.controller && room.controller.my).controller.owner.username`

Comment (Nevrdid) [inline src/config_brain_memory.js]: all way : 
http://support.screeps.com/hc/en-us/articles/115001774029 ->
`Added new constants SYSTEM_USERNAME, SIGN_NOVICE_AREA, SIGN_RESPAWN_AREA. `

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 110/336
    Coefficient: 0.327380952381
    Merging in 6 days 15 hours
    Age 0 days 2 hours

File: src/config_brain_memory.js (hunk 1 above)
Comment (karlthepagan) [inline src/config_brain_memory.js]: SYSTEM_USERNAME is "Screeps"

Regarding performance panic over using chain, search for "lodash shortcut fusion" for example: http://gajus.com/blog/4/harder-better-faster-stronger-lo-dash-v3#shortcut-fusion

Comment (Nevrdid) [inline src/config_brain_memory.js]: we don't need to chain, only a find is enouth. plus using filter as you did will all ways not return usernam until all rooms are looped, chaining or not. use only find allow to stop at first found

**Pull request merged**